twisted/conch/ssh/transport.py

```diff
@@ -1499,6 +1598,59 @@ def ssh_KEXINIT(self, packet):
                     self._dhMaximalGroupSize,
                     ))
 
+    def _ssh_KEX_ECDH_REPLY(self, packet):
+        """
+        Called to handle a reply to a ECDH exchange message(KEX_ECDH_INIT).
+
+        Like the handler for I{KEXDH_INIT}, this message type has an
+        overlapping value.  This method is called from C{ssh_KEX_DH_GEX_GROUP}
+        if that method detects a non-group key exchange is in progress.
+
+        Payload::
+
+            string serverHostKey
+            string server Elliptic Curve Diffie-Hellman public key
+            string signature
+
+        We verify the host key and continue if it passes verificiation.
+        Otherwise raise an exception and return.
+
+        @type packet: L{bytes}
+        @param packet: The message data.
+
+        @return: None.
+        """
+        #Get the host public key
+        self.theirECHost, packet = getNS(packet)
+
+        #Get the raw ECDH public key bytes
+        pktPub, packet = getNS(packet)
+
+        #Get the signature
+        signature, packet = getNS(packet)
+
+        h = _kex.getHashProcessor(self.kexAlg)()
+        h.update(NS(self.ourVersionString))
+        h.update(NS(self.otherVersionString))
+        h.update(NS(self.ourKexInitPayload))
+        h.update(NS(self.otherKexInitPayload))
+        h.update(NS(self.theirECHost))
+
+        #Take the provided public key and transform it into a format for the cryptography module
+        self.theirECPub = ec.EllipticCurvePublicNumbers.from_encoded_point(self.curve, pktPub).public_key(default_backend())
```

Comment:
The breaking up of updating the hash is a left over of my experimentation, but I agree it should be changed.

Raising InvalidSignature was to make troubleshooting a little easier by identifying what went wrong and where as close to the exception as possible.  This was a big headache in implementing ECDH so I thought I'd be nice and help the next poor sap who had the same trouble I did. :) 

theirECHost has to exist, but not as self.theirECHost.  I was following the convention I saw else where of defining parts of the key were saved as self variables (e.g. self.x and self.g from ssh_KEXINIT).

Everything I mention in the comment isn't necessarily required so I'm happy to change what you see fit.